### PR TITLE
feature: add get_block_finalized_reward_info rpc

### PR DESCRIPTION
**Pull request by u2**
Firstly fix test:  `cellbase` should have `witness`, otherwise the RPC test will panic.

https://github.com/nervosnetwork/ckb/pull/1297/commits/25445bc4ab1c35e160fefe2bacd89ada32fcda76

### CODE REVIEW

File: util/reward-calculator/src/lib.rs [hunk 1]
```diff
@@ -33,7 +34,10 @@ impl<'a, P: ChainProvider> RewardCalculator<'a, P> {
 
     /// `RewardCalculator` is used to calculate block finalize target's reward according to the parent header.
     /// block reward consists of four parts: base block reward, tx fee, proposal reward, and secondary block reward.
-    pub fn block_reward(&self, parent: &Header) -> Result<(Script, Capacity), FailureError> {
+    pub fn block_reward(
+        &self,
+        parent: &Header,
+    ) -> Result<(Script, Capacity, BlockReward), FailureError> {
```
Comment (doitian) [inline util/reward-calculator/src/lib.rs]: We can impl a method on `BlockReward` to return the total capacity, then we don't need to return the total capacity here.

Comment (u2) [reply util/reward-calculator/src/lib.rs]: or add a new field `total`?

File: rpc/src/module/chain.rs [hunk 2]
```diff
@@ -251,4 +254,17 @@ impl<CS: ChainStore + 'static> ChainRpc for ChainRpcImpl<CS> {
...
     }
+
+    fn get_block_finalized_reward_info(&self, hash: H256) -> Result<Option<BlockRewardView>> {
+        Ok(self
+            .shared
+            .store()
+            .get_block_header(&hash)
+            .and_then(|header| {
+                self.shared
+                    .finalize_block_reward(&header)
```
Comment (doitian) [inline rpc/src/module/chain.rs]: I'm afraid `finalize_block_reward` requires parent hash instead of the current block hash.

Comment (doitian) [reply rpc/src/module/chain.rs]: The reason is that the info is required when constructing the new block, which hash is not available yet.

Comment (doitian) [reply rpc/src/module/chain.rs]: Maybe we should rename `finalize_block_reward` to more verbose `finalize_block_reward_via_parent_hash`, and `RewardCalculator::block_reward` to `RewardCalculator::block_reward_via_parent_hash`.

File: rpc/json/rpc.json [hunk 3]
```diff
@@ -666,6 +666,25 @@
             }
         ]
     },
+    {
+        "description": "Returns the information about block reward.",
```
Comment (doitian) [inline rpc/json/rpc.json]:
```suggestion
        "description": "Returns each component of the created CKB in this block's cellbase, which is issued to a block N - 1 - ProposalWindow.farthest, where this block's height is N",
```

**Comment by doitian** [comment]
Suggest a new name `get_cellbase_output_capacity_details`

File: rpc/src/module/chain.rs (hunk 2 above)
Comment (u2) [reply rpc/src/module/chain.rs]: `fn finalize_block_reward(&self, _parent: &Header)`  implicitly tell that.

File: rpc/json/rpc.json (hunk 3 above)
Comment (u2) [reply rpc/json/rpc.json]: thanks

File: rpc/src/module/chain.rs (hunk 2 above)
Comment (u2) [reply rpc/src/module/chain.rs]: updated, would find the parent first and then finalize_block_reward.

File: rpc/README.md [hunk 4]
```diff
@@ -233,7 +246,45 @@ http://localhost:8114
...
+```json
+{
+    "id": 2,
+    "jsonrpc": "2.0",
+    "result": {
+        "primary": "100000000000",
+        "proposal_reward": "0",
+        "secondary": "3936000",
+        "total": "100003936000",
+        "tx_fee": "0"
```
Comment (keroro520) [inline rpc/README.md]: Why `proposal_reward` has a suffix but others not?

What about:

`primary_block_reward`
`secondary_block_reward`
`total_reward`
`transactions_fee`

Comment (u2) [reply rpc/README.md]: > Why `proposal_reward` has a suffix but others not?
> 
> What about:
> 
> `primary_block_reward`
> `secondary_block_reward`
> `total_reward`
> `transactions_fee`

https://github.com/cryptape/ckb-internal/issues/1206

**Pull request merged**